src/main/java/io/kestra/plugin/ai/retriever/SqlDatabaseRetriever.java

```diff
@@ -0,0 +1,231 @@
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.ai.domain.ContentRetrieverProvider;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import javax.sql.DataSource;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.logging.Logger;
+
+@Getter
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonDeserialize
+@Schema(
+    title = "SQL Database content retriever using LangChain4j experimental SqlDatabaseContentRetriever. " +
+        "⚠ IMPORTANT: the database user should have READ-ONLY permissions."
+)
+@Plugin(
+    examples = {
+        @Example(
+            full = true,
+            title = "RAG chat with a SQL Database content retriever (answers grounded in database data)",
+            code = """
+                id: rag
+                namespace: company.ai
+
+                tasks:
+                  - id: chat_with_rag_and_sql_retriever
+                    type: io.kestra.plugin.ai.rag.ChatCompletion
+                    chatProvider:
+                      type: io.kestra.plugin.ai.provider.OpenAI
+                      modelName: gpt-4o-mini
+                      apiKey: "{{ kv('OPENAI_API_KEY') }}"
+                    contentRetrievers:
+                      - type: io.kestra.plugin.ai.retriever.SqlDatabaseRetriever
+                        jdbcUrl: "jdbc:postgresql://localhost:5432/mydb"
+                        username: "{{ kv('DB_USER') }}"
+                        password: "{{ kv('DB_PASSWORD') }}"
+                        driver: "org.postgresql.Driver"
+                        provider: "OPENAI"
+                        modelName: "gpt-4o-mini"
+                        apiKey: "{{ kv('OPENAI_API_KEY') }}"
+                    prompt: "What are the top 5 customers by revenue?"
+                """
+        )
+    }
+)
+public class SqlDatabaseRetriever extends ContentRetrieverProvider {
+
+    public enum Provider {
+        OPENAI,
+        AZURE_OPENAI,
+        GEMINI,
+        ANTHROPIC,
+        MISTRAL,
+        OLLAMA,
+        QWEN,
+        VERTEX_GEMINI,
+        VERTEX,
+        WORKERS,
+        BEDROCK,
+        LOCAL
+    }
+
+    @Schema(title = "JDBC connection URL to the target database.")
+    @NotNull
+    private Property<String> jdbcUrl;
+
+    @Schema(title = "Database username.")
+    @NotNull
+    private Property<String> username;
+
+    @Schema(title = "Database password.")
+    @NotNull
+    private Property<String> password;
+
+    @Schema(title = "Fully qualified JDBC driver class name (e.g. org.postgresql.Driver, com.mysql.cj.jdbc.Driver).")
+    @NotNull
+    private Property<String> driver;
+
+    @Schema(title = "LLM provider.")
+    @NotNull
+    private Property<Provider> provider;
```

Comment:
Why not use our `ModelProvider` as done in the `AiAgent` tool?

Reply:
By using our own ModelProvider users would be able to set all supported properties instead of just apikey and model name.